docs/data/data-grid/row-recipes/row-recipes.md

```diff
@@ -0,0 +1,15 @@
+---
+title: Data Grid - Row customization recipes
+---
+
+# Data Grid - Row customization recipes
+
+<p class="description">Advanced row customization recipes.</p>
+
+## One expanded detail panel at a time
```

Comment:
Another option was to add this example to the [Master detail docs](https://mui.com/x/react-data-grid/master-detail/), but I decided to avoid increasing the size of the feature doc page and add it here instead.